chicken-sloths/bangazon-cli-sprint2
# Dlk pay type ctrl

# Description
This completes the feature: closing active orders. All the different functionality is connected in the `completeOrderC`. Different cases - like the customer not having an active order, or the user canceling the completion process - are handled.

## Related Ticket(s)
Fixes #6 

## Problem to Solve
Users should be able to close active orders for a customer.

## Proposed Changes
This PR completes the functionality in `completeOrderC`, which connects `completeOrderV`, `OrdersM`, `PaymentOptionsM`, printing out a prompt leading the user to close out an order

## Expected Behavior
If the user selects a customer with an active order and payment options, selects one of those payment options, then the active should become closed. I.e., the order now has a payment option associated with it. That customer will now have no active orders and the user should be kicked back to the main menu if they select `6` again (before opening a new order).
If the customer doesn't have available payment options, then the user is kicked back to the main menu.
If the user selects `N` then the user is kicked back to the main menu
If the customer has no active orders, then the user is kicked back to the main menu.

## Steps to Test Solution

1. `npm db:reset` <- essential since there is new fake data
1. `npm test`, all tests should pass
1. `npm start`
1. Select `2` (choose active customer)
1. Select `3` (customer 3)
1. Select `6` (close active order: total should be $3377)
1. Select `N` (to cancel, should return to main menu)
1. Select `6` (close active order: total should be $3377)
1. Select `Y` (payment options should be: 1. Visa, 2. MasterCard)
1. Select `1` or `2`
1. `Order 2 closed` should print, followed by the main menu!
NOTE: This actually changes the DB. To reset that change, type this in your DB browser
```
UPDATE Orders SET payment_option_id=null WHERE customer_id=3
```

## Testing
